core/utils/src/main/kotlin/edu/stanford/spezi/core/utils/extensions/SemanticsPropertyKey.kt

```diff
@@ -0,0 +1,5 @@
+package edu.stanford.spezi.core.utils.extensions
+
+import androidx.compose.ui.semantics.SemanticsPropertyKey
+
+val ImageResourceKey = SemanticsPropertyKey<String>("ImageResourceKey")
```

Comment:
Let's have a full fledged setup for this 🙌 

We can add the following extension function in `Modifier.kt` in utils module:

```kotlin
fun Modifier.imageResourceIdentifier(identifier: String) = testTag(tag = identifier)
    .semantics { this[ImageResourceKey] = identifier }
```

It adds by default the tag and the image resource key. This allows replacing the usages to: `modifier = Modifier.imageResourceIdentifier(imageResource.identifier)`

Regarding testing, you can add a new file in `testing` module with this extension:

```kotlin
fun SemanticsNodeInteraction.assertImageIdentifier(identifier: String) = assert(
    SemanticsMatcher.expectValue(ImageResourceKey, identifier)
)
```

These will result in:

```kotlin
@Composable
fun ImageResourceIcon(
    imageResource: ImageResource,
    contentDescription: String,
    modifier: Modifier = Modifier,
    tint: Color = Colors.primary,
) {
    val imageModifier = modifier.then(Modifier.imageResourceIdentifier(imageResource.identifier))
    when (imageResource) {
        is ImageResource.Vector -> {
            Icon(
                imageVector = imageResource.image,
                contentDescription = contentDescription,
                tint = tint,
                modifier = imageModifier
            )
        }

        is ImageResource.Drawable -> {
            Icon(
                painter = painterResource(id = imageResource.resId),
                contentDescription = contentDescription,
                tint = tint,
                modifier = imageModifier
            )
        }
    }
}
```

And in tests you assert as follows:

```kotlin
fun assertHasImage(imageResource: ImageResource?) {
    imageResource?.let {
        image(imageResource)
            .assertExists()
            .assertContentDescriptionContains("Profile Picture")
            .assertImageIdentifier(it.identifier)
    }
}
```